File: tasks_26/solutions/26-46.pas
// Автор: А.М. Кабанов

##
var F:= openRead('26-46.txt');
var n:= F.ReadInteger;
var a:= (1..n).Select(i->F.ReadInteger).Sorted.ToArray;
 
var (count, m):=(0, longInt.MaxValue);
for var i:=0 to a.High-2 do
  for var j:=i+1 to a.High-1 do
    for var k:=j+1 to a.High do begin
      var s:= a[i]+a[j]+a[k];
      if s mod 3 = 0 then begin
        var x:= s div 3;
        if a.BinarySearch(x)>=0 then begin
          count:=count+1;
          m:=min(m,x);
        end;
      end;
    end;
print(count,m);

File: tasks_26/solutions/26-46.py
with open('26-46.txt') as F:
  N = int(F.readline())
  data = [int(s) for s in F]

who = dict( zip(data, [1 for i in range(N)]) )

count, mi = 0, 10**10
for i in range(N-2):
  for j in range(i+1,N-1):
    for k in range(j+1,N):
      s = data[i] + data[j] + data[k]
      if s % 3 == 0 and s//3 in who:
        count += 1
        mi = min( mi, s//3 )

print(count, mi)
